src/main/java/de/tum/in/test/api/ast/model/JavaFile.java

```diff
@@ -26,11 +28,22 @@ public class JavaFile {
 	private final Path javaFilePath;
 	private final CompilationUnit javaFileAST;
 
-	public JavaFile(Path javaFilePath, CompilationUnit javaFileAST) {
+	public JavaFile(Path javaFilePath, CompilationUnit javaFileAST, boolean excludeMainMethod) {
 		this.javaFilePath = javaFilePath;
+		if (excludeMainMethod) {
+			excludeMainMethod(javaFileAST);
+		}
 		this.javaFileAST = javaFileAST;
 	}
 
+	private static void excludeMainMethod(CompilationUnit javaFileAST) {
+		javaFileAST.findAll(MethodDeclaration.class).stream()
+				.filter(method -> method.isStatic() && method.getNameAsString().equals("main")
+						&& method.getParameters().size() == 1 && method.getType().isVoidType()
+						&& method.getParameter(0).getTypeAsString().equals("String[]"))
```

Comment:
If I'm not mistaken a `public static void main()` (Without any arguments) is also a valid Java entry point.

Reply:
No, at least not yet. But this will change in the future ( JEP-477 ). Note, however, that a good portion of the Ares functionality is not compatible with newer JDKs anyway.

Reply:
This is not in scope of this current PR, will adress this issue in another PR according to my thesis.